src/gui/qgsadvanceddigitizingdockwidget.h

```diff
@@ -271,6 +276,36 @@ class GUI_EXPORT QgsAdvancedDigitizingDockWidget : public QgsDockWidget, private
      */
     bool alignToSegment( QgsMapMouseEvent *e, QgsAdvancedDigitizingDockWidget::CadConstraint::LockMode lockMode = QgsAdvancedDigitizingDockWidget::CadConstraint::HardLock );
 
+    /**
+     * Processes the canvas press \a event.
+     * \returns Returns TRUE if the event has been 'eaten' and should not propagate further.
```

Comment:
As noted before, it'd be cleaner to use event->isAccepted()/isIgnored() here to determine propagation.